### PR TITLE
Use std::make_xxx

### PR DESCRIPTION
**Pull request by baldersheim**
nested namespace
And various other style changes.
@toregge PR

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/attribute/attribute_directory.h [hunk 1]
```diff
@@ -50,6 +49,9 @@ class AttributeDirectory
                        const vespalib::string &name);
     ~AttributeDirectory();
 
+    const vespalib::string & getAttrName() const { return _name; }
+    vespalib::string getSnapshotDir(SerialNum serialNum);
```
Comment (toregge) [inline searchcore/src/vespa/searchcore/proton/attribute/attribute_directory.h]: getSnapshotDir() should only be callable via nested Writer class (to ensure it isn't removed by another thread while we're using it), or via getAttributeFileName() (when loading attribute at startup).

File: searchcore/src/vespa/searchcore/proton/attribute/attribute_initializer.cpp [hunk 2]
```diff
@@ -206,30 +192,27 @@ AttributeInitializer::loadAttribute(const AttributeVectorSP &attr,
...
     if (serialNum < _currentSerialNum) {
-        logAttributeTooOld(attr, serialNum, _currentSerialNum);
+        logAttributeTooOld(header, serialNum, _currentSerialNum);
     }
     if (!headerTypeOK(header, attr->getConfig())) {
         logAttributeWrongType(attr, header);
     }
-    LOG(info, "Returning empty attribute vector for '%s'",
-            attr->getBaseFileName().c_str());
+    LOG(info, "Returning empty attribute vector for '%s'", attr->getName().c_str());
```
Comment (toregge) [inline searchcore/src/vespa/searchcore/proton/attribute/attribute_initializer.cpp]: Too little information (document type and document subdb name can no longer be deduced from log message).

File: searchcore/src/vespa/searchcore/proton/attribute/attributemanager.cpp [hunk 3]
```diff
@@ -145,7 +145,7 @@ AttributeManager::internalAddAttribute(const AttributeSpec &spec,
 void
 AttributeManager::addAttribute(const AttributeWrap &attribute, const ShrinkerSP &shrinker)
 {
-    LOG(debug, "Adding attribute vector '%s'", attribute.getAttribute()->getBaseFileName().c_str());
+    LOG(debug, "Adding attribute vector '%s'", attribute.getAttribute()->getName().c_str());
```
Comment (toregge) [inline searchcore/src/vespa/searchcore/proton/attribute/attributemanager.cpp]: Too little information, document type and document subdb name cannot be deduced from log message.

**Review by toregge — APPROVED** [review]
Looks mostly OK.
Some information loss due to changed log messages.

File: searchcore/src/vespa/searchcore/proton/attribute/attribute_directory.h (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/attribute/attribute_directory.h]: I moved it back to private scope. It was not needed in this PR anyway.

File: searchcore/src/vespa/searchcore/proton/attribute/attribute_initializer.cpp (hunk 2 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/attribute/attribute_initializer.cpp]: I pulled this one back. It will be redone later on.

**Pull request merged**